### PR TITLE
added better type checks

### PR DESCRIPTION
**Pull request by jameslamb**
In this PR, I improved some of our assertions in the R package.

So e.g. `data.table::is.data.table(thingDT)` is preferred to `"data.table" %in% class(thingDT)`.

### CODE REVIEW

File: r-pkg/R/es_search.R [hunk 1]
```diff
@@ -568,15 +568,15 @@ es_search <- function(es_host
...
+    if (! is.character(es_host)){
+        msg <- paste0("es_host should be a string. You gave an object of type"
                       , paste0(class(es_host), collapse = '/'))
         log_fatal(msg)
     }
 
     # [2] es_host is length 1
     if (! length(es_host) == 1){
-        msg <- paste0("es_host should be length 1!"
+        msg <- paste0("es_host should be length 1."
```
Comment (austin3dickey) [inline r-pkg/R/es_search.R]: haha

Comment (jameslamb) [reply r-pkg/R/es_search.R]: so important

Comment (jameslamb) [reply r-pkg/R/es_search.R]: it felt like we were being too aggressive

**Comment by jameslamb** [comment]
> Looks good! All the R tests are failing though. Travis doesn't print the tracebacks so it's hard to tell why.

I'm sure I messed up something silly. Will take a look

**Comment by jameslamb** [comment]
> > Looks good! All the R tests are failing though. Travis doesn't print the tracebacks so it's hard to tell why.
> 
> I'm sure I messed up something silly. Will take a look

1. I hate the way that commenting works on GitHub, where sometimes you can thread and sometimes you have to quote-reply
2. When I ran the tests locally, I was able to get the same tests to fail by not copying the test data into `r-pkg/inst/testdata`. I'm going to kick off a build where I directly run the tests (instead of `R CMD CHECK`) so maybe we can get more logs

File: r-pkg/R/es_search.R [hunk 2]
```diff
@@ -85,7 +85,7 @@ es_search <- function(es_host
 ){
 
     # Check if this is an aggs or straight-up search query
-    if (length(query_body) > 1 || ! "character" %in% class(query_body)){
+    if (assertthat::is.string(query_body)){
```
Comment (jameslamb) [inline r-pkg/R/es_search.R]: this assertion was the problem with Travis! I should have made it `!is.string()` and I made it `is.string()`. Good for us for having good tests!

Comment (jameslamb) [reply r-pkg/R/es_search.R]: yeah this is working now 😄

**Pull request merged**